python/doc/examples/probabilistic_modeling/discrete_markov_chain_process.ipynb

```diff
@@ -0,0 +1,200 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create a discrete Markov chain process"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This example details first how to create and manipulate a discrete Markov chain.\n",
+    "\n",
+    "A discrete Markov chain $X: \\Omega \\times \\mathcal{D} \\rightarrow E$, where $E = [\\![ 0,...,n]\\!]$ is a process\n",
+    "where $\\mathcal{D}=\\mathbb{R}$ discretized on the time grid $(t_k)_{k \\geq 0}$ such\n",
+    "that:\n",
+    "\n",
+    "$$\\begin{aligned}\n",
+    "      \\forall k>0,\\: \\mathbb{P} ( X_{t_k} | X_{t_0},...X_{t_{k-1}} )  =  \\mathbb{P} ( X_{t_k} | X_{t_{k-1}} )\n",
+    "  \\end{aligned}\n",
+    "$$\n",
+    "\n",
+    "The transition matrix of the process $\\mathcal{M} = (m_{i,j})$ can be defined such that:\n",
+    "\n",
+    "$$\\begin{aligned}\n",
+    "     \\forall t_k \\in \\mathcal{D}, \\forall i,j \\leq n, m_{i,j} = \\mathbb{P} (X_{t_{k+1}} = j | X_{t_{k}} = i)\n",
+    "  \\end{aligned}$$\n",
+    "\n",
+    "The library proposes to model it through the object *DiscreteMarkovChain* defined thanks to the origin $X_{t_0}$, the transition matrix $\\mathcal{M}$ and the time grid."
```

Comment:
The origin is deterministic in the implementation. It should not be in capital letter. Otherwise one could allow for a random origin, specified through an integral 1D distribution with bounded support (and with a Dirac distribution for a deterministic origin)